atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -1081,102 +1084,109 @@ private void fillOverflowValues(DbReadTable table,
 
     @Override
     public void truncateTable(TableReference tableRef) {
-        runDdl(tableRef, new Function<DbDdlTable, Void>() {
-            @Override
-            public Void apply(DbDdlTable table) {
-                table.truncate();
-                return null;
-            }
-        });
+        truncateTables(ImmutableSet.of(tableRef));
+    }
+
+    @Override
+    public void truncateTables(Set<TableReference> tableRefs) {
+        List<Future> futures = Lists.newArrayListWithCapacity(tableRefs.size());
+        tableRefs.forEach(tableRef -> futures.add(ddlExecutor.submit(() ->
```

Comment:
yeah, I thought about it but couldn't find a great way to deal with Set elements vs. Map Entry elements in the underlying logic that didn't look gross.